homeassistant/components/apple_tv/media_player.py

```diff
@@ -239,12 +253,16 @@ def _is_feature_available(self, feature):
 
     async def async_turn_on(self):
         """Turn the media player on."""
-        await self.manager.connect()
+        if self._is_feature_available(FeatureName.TurnOn):
+            await self.atv.power.turn_on()
 
     async def async_turn_off(self):
         """Turn the media player off."""
-        self._playing = None
-        await self.manager.disconnect()
+        if (self._is_feature_available(FeatureName.TurnOff)) and (
+            not self._is_feature_available(FeatureName.PowerState)
```

Comment:
Any reason why you check this here and not in `turn_on` (or vice versa)?

Reply:
Yes, in turn on - if it will be executed when it's already on, nothing will happen
But, in turn off, if turn off will be executed and the state is already off it will turn it on (while we want to keep the tv off)